compiler/ballerina-backend-jvm-old/src/main/ballerina/src/compiler_backend_jvm/jvm_method_gen.bal

```diff
@@ -680,13 +683,21 @@ function generateBasicBlocks(jvm:MethodVisitor mv, bir:BasicBlock?[] basicBlocks
         currentEE = errorEntries[errorEntryCnt];
     }
 
+    int caseIndex = 0;
+
     while (j < basicBlocks.length()) {
         bir:BasicBlock bb = getBasicBlock(basicBlocks[j]);
         string currentBBName = io:sprintf("%s", bb.id.value);
 
         // create jvm label
         jvm:Label bbLabel = labelGen.getLabel(funcName + bb.id.value);
         mv.visitLabel(bbLabel);
+        if (j == 0 && !isArg) {
+            // SIPUSH range is (-32768 to 32767) so if the state index goes beyond that, need to use visitLdcInsn
```

Comment:
Why don't we use visitLdcInsn now itself?

Reply:
I think we need to revisit that later(depending on the requirements) main reason not to use lcd now is I think to reduce bytes. (BTW this is a copy-pasted section from below,  along with the comment :) )